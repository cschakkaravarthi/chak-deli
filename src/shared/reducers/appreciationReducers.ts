import { Reducer } from 'redux';

import {
  AppreciationState,
  FETCH_APPRECIATION_DATA,
  LOAD_APPRECIATION_ALERT,
  CLEAR_APPRECIATION_REDUCER,
  LOAD_APPRECIATION_ALERT_ERROR,
  LOAD_APPRECIATION_MANAGER_ALERT,
  SET_NEW_CARD_DETAIL,
  FETCH_CARD_TEMPLATES,
  CLEAR_NEW_CARD_FORM,
  TOGGLE_SUCCESS_SENT_ALERT,
  GET_APPRECIATION_CARDS_LIST_GROUPED,
  VIEW_APPRECIATION_CARD,
  ARCHIVE_ALERT_STATE,
  APPRECIATION_CARD_ID,
  LOAD_CARDS_IN_GROUP
} from '../types/appreciationTypes';

export const initialState = {
  cardId: '',
  isFetching: false,
  appreciationCardAlert: undefined,
  appreciationCardAlertError: false,
  appreciationManagerCardAlert: undefined,
  newAppreciationCardForm: {
    templateId: '',
    recipientEmail: '',
    recipientFirstName: '',
    recipientLastName: '',
    cardMessage: '',
    recipientSupervisorEmail: '',
    managerName: ''
  },
  cardTemplates: [],
  showSentSucessAlert: false,
  appreciationCardsListGrouped: {
    sent: [],
    received: [],
    team: []
  },
  cardToView: undefined,
  displayCardView: false,
  cardArchiveSuccess: false
};

const appreciationReducers: Reducer<AppreciationState, any> = (state = initialState, action) => {
  switch (action.type) {
    case APPRECIATION_CARD_ID:
      return {
        ...state,
        cardId: action.cardId
      };
    case ARCHIVE_ALERT_STATE:
      return {
        ...state,
        cardArchiveSuccess: action.cardArchiveSuccess
      };
    case FETCH_APPRECIATION_DATA:
      return {
        ...state,
        isFetching: action.isFetching
      };
    case LOAD_APPRECIATION_ALERT:
      return {
        ...state,
        isFetching: action.isFetching,
        appreciationCardAlert: action.appreciationCardAlert,
        appreciationCardAlertError: false
      };
    case LOAD_APPRECIATION_MANAGER_ALERT:
      return {
        ...state,
        isFetching: action.isFetching,
        appreciationManagerCardAlert: action.appreciationManagerCardAlert,
        appreciationCardAlertError: false
      };
    case LOAD_APPRECIATION_ALERT_ERROR:
      return {
        ...state,
        isFetching: action.isFetching,
        appreciationCardAlertError: true
      };
    case CLEAR_APPRECIATION_REDUCER:
      return {
        ...state,
        isFetching: initialState.isFetching,
        appreciationCardAlert: initialState.appreciationCardAlert,
        appreciationCardAlertError: initialState.appreciationCardAlertError,
        appreciationManagerCardAlert: initialState.appreciationManagerCardAlert
      };
    case SET_NEW_CARD_DETAIL:
      return {
        ...state,
        newAppreciationCardForm: {
          ...state.newAppreciationCardForm,
          ...action.payload
        }
      };
    case FETCH_CARD_TEMPLATES:
      return {
        ...state,
        cardTemplates: action.cardTemplates
      };
    case CLEAR_NEW_CARD_FORM:
      return {
        ...state,
        newAppreciationCardForm: initialState.newAppreciationCardForm
      };
    case TOGGLE_SUCCESS_SENT_ALERT:
      return {
        ...state,
        showSentSucessAlert: action.showSentSucessAlert
      };
    case GET_APPRECIATION_CARDS_LIST_GROUPED:
      return {
        ...state,
        appreciationCardsListGrouped: action.appreciationCardsListGrouped
      };
    case VIEW_APPRECIATION_CARD:
      return {
        ...state,
        cardToView: action.displayCardView ? action.cardToView : initialState.cardToView,
        displayCardView: action.displayCardView
      };
    case LOAD_CARDS_IN_GROUP:
      return {
        ...state,
        appreciationCardsListGrouped: {
          ...state.appreciationCardsListGrouped,
          [action.cardsType]: action.appreciationCards
        }
      };
    default:
      return state;
  }
};

export default appreciationReducers;
