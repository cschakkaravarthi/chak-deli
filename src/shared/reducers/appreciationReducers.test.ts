import appreciationReducers, { initialState } from './appreciationReducers';

import {
  SET_NEW_CARD_DETAIL,
  ARCHIVE_ALERT_STATE,
  APPRECIATION_CARD_ID,
  LOAD_APPRECIATION_ALERT,
  CLEAR_APPRECIATION_REDUCER,
  LOAD_APPRECIATION_ALERT_ERROR
} from '../types/appreciationTypes';

import { dummyAppreciationCardNotification } from '../models/Notification.model';

describe('appreciationReducers', () => {
  it('Handles default case', () => {
    const state = appreciationReducers(undefined, {});
    expect(state).toEqual(initialState);
  });

  it('Handles APPRECIATION_CARD_ID action', () => {
    const action = { type: APPRECIATION_CARD_ID, cardId: '1' };
    const state = appreciationReducers(initialState, action);
    expect(state).toMatchSnapshot();
  });

  it('Handles ARCHIVE_ALERT_STATE action', () => {
    const action = { type: ARCHIVE_ALERT_STATE, cardArchiveSuccess: true };
    const state = appreciationReducers(initialState, action);
    expect(state).toMatchSnapshot();
  });

  it('Handles LOAD_APPRECIATION_ALERT action', () => {
    const action = { type: LOAD_APPRECIATION_ALERT, appreciationCardAlert: [dummyAppreciationCardNotification] };
    const state = appreciationReducers(initialState, action);
    expect(state).toMatchSnapshot();
  });

  it('Handles LOAD_APPRECIATION_ALERT_ERROR action', () => {
    const action = { type: LOAD_APPRECIATION_ALERT_ERROR, appreciationCardAlertError: true };
    const state = appreciationReducers(initialState, action);
    expect(state).toMatchSnapshot();
  });

  it('Handles CLEAR_APPRECIATION_REDUCER action', () => {
    const action = {
      type: CLEAR_APPRECIATION_REDUCER,
      appreciationCardAlert: initialState.appreciationCardAlert,
      appreciationCardAlertError: initialState.appreciationCardAlertError
    };

    const state = appreciationReducers(initialState, action);
    expect(state).toMatchSnapshot();
  });

  it('Handles SET_NEW_CARD_DETAIL action', () => {
    const action = {
      type: SET_NEW_CARD_DETAIL,
      payload: { cardMessage: 'Card Message' }
    };

    const state = appreciationReducers(initialState, action);
    expect(state).toMatchSnapshot();
  });
});
