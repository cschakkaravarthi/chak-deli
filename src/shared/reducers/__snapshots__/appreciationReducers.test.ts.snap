// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appreciationReducers Handles APPRECIATION_CARD_ID action 1`] = `
Object {
  "appreciationCardAlert": undefined,
  "appreciationCardAlertError": false,
  "appreciationCardsListGrouped": Object {
    "received": Array [],
    "sent": Array [],
    "team": Array [],
  },
  "appreciationManagerCardAlert": undefined,
  "cardArchiveSuccess": false,
  "cardId": "1",
  "cardTemplates": Array [],
  "cardToView": undefined,
  "displayCardView": false,
  "isFetching": false,
  "newAppreciationCardForm": Object {
    "cardMessage": "",
    "managerName": "",
    "recipientEmail": "",
    "recipientFirstName": "",
    "recipientLastName": "",
    "recipientSupervisorEmail": "",
    "templateId": "",
  },
  "showSentSucessAlert": false,
}
`;

exports[`appreciationReducers Handles ARCHIVE_ALERT_STATE action 1`] = `
Object {
  "appreciationCardAlert": undefined,
  "appreciationCardAlertError": false,
  "appreciationCardsListGrouped": Object {
    "received": Array [],
    "sent": Array [],
    "team": Array [],
  },
  "appreciationManagerCardAlert": undefined,
  "cardArchiveSuccess": true,
  "cardId": "",
  "cardTemplates": Array [],
  "cardToView": undefined,
  "displayCardView": false,
  "isFetching": false,
  "newAppreciationCardForm": Object {
    "cardMessage": "",
    "managerName": "",
    "recipientEmail": "",
    "recipientFirstName": "",
    "recipientLastName": "",
    "recipientSupervisorEmail": "",
    "templateId": "",
  },
  "showSentSucessAlert": false,
}
`;

exports[`appreciationReducers Handles CLEAR_APPRECIATION_REDUCER action 1`] = `
Object {
  "appreciationCardAlert": undefined,
  "appreciationCardAlertError": false,
  "appreciationCardsListGrouped": Object {
    "received": Array [],
    "sent": Array [],
    "team": Array [],
  },
  "appreciationManagerCardAlert": undefined,
  "cardArchiveSuccess": false,
  "cardId": "",
  "cardTemplates": Array [],
  "cardToView": undefined,
  "displayCardView": false,
  "isFetching": false,
  "newAppreciationCardForm": Object {
    "cardMessage": "",
    "managerName": "",
    "recipientEmail": "",
    "recipientFirstName": "",
    "recipientLastName": "",
    "recipientSupervisorEmail": "",
    "templateId": "",
  },
  "showSentSucessAlert": false,
}
`;

exports[`appreciationReducers Handles LOAD_APPRECIATION_ALERT action 1`] = `
Object {
  "appreciationCardAlert": Array [
    Object {
      "cardId": "40518cda-f3eb-406f-904f-7f256def2b2f",
      "cardMessage": "Your so cute Rudy",
      "dateSent": "1576881752255",
      "recipientArchived": false,
      "recipientEmail": "marion1.wirsig1@umusic.com",
      "recipientFirstName": "Marion",
      "recipientLastName": "Wirsig",
      "recipientSupervisorEmail": "meena1.belagam1@umusic.com",
      "senderEmail": "meena.belagam@umusic.com",
      "senderFirstName": "Meena",
      "senderLastName": "Belagam",
      "supervisorArchived": false,
    },
  ],
  "appreciationCardAlertError": false,
  "appreciationCardsListGrouped": Object {
    "received": Array [],
    "sent": Array [],
    "team": Array [],
  },
  "appreciationManagerCardAlert": undefined,
  "cardArchiveSuccess": false,
  "cardId": "",
  "cardTemplates": Array [],
  "cardToView": undefined,
  "displayCardView": false,
  "isFetching": undefined,
  "newAppreciationCardForm": Object {
    "cardMessage": "",
    "managerName": "",
    "recipientEmail": "",
    "recipientFirstName": "",
    "recipientLastName": "",
    "recipientSupervisorEmail": "",
    "templateId": "",
  },
  "showSentSucessAlert": false,
}
`;

exports[`appreciationReducers Handles LOAD_APPRECIATION_ALERT_ERROR action 1`] = `
Object {
  "appreciationCardAlert": undefined,
  "appreciationCardAlertError": true,
  "appreciationCardsListGrouped": Object {
    "received": Array [],
    "sent": Array [],
    "team": Array [],
  },
  "appreciationManagerCardAlert": undefined,
  "cardArchiveSuccess": false,
  "cardId": "",
  "cardTemplates": Array [],
  "cardToView": undefined,
  "displayCardView": false,
  "isFetching": undefined,
  "newAppreciationCardForm": Object {
    "cardMessage": "",
    "managerName": "",
    "recipientEmail": "",
    "recipientFirstName": "",
    "recipientLastName": "",
    "recipientSupervisorEmail": "",
    "templateId": "",
  },
  "showSentSucessAlert": false,
}
`;

exports[`appreciationReducers Handles SET_NEW_CARD_DETAIL action 1`] = `
Object {
  "appreciationCardAlert": undefined,
  "appreciationCardAlertError": false,
  "appreciationCardsListGrouped": Object {
    "received": Array [],
    "sent": Array [],
    "team": Array [],
  },
  "appreciationManagerCardAlert": undefined,
  "cardArchiveSuccess": false,
  "cardId": "",
  "cardTemplates": Array [],
  "cardToView": undefined,
  "displayCardView": false,
  "isFetching": false,
  "newAppreciationCardForm": Object {
    "cardMessage": "Card Message",
    "managerName": "",
    "recipientEmail": "",
    "recipientFirstName": "",
    "recipientLastName": "",
    "recipientSupervisorEmail": "",
    "templateId": "",
  },
  "showSentSucessAlert": false,
}
`;
